docs/how-to-guides/effective-gno.md

```diff
@@ -0,0 +1,255 @@
+---
+id: 'effective-gno'
+---
+
+# Effective Gno
+
+Welcome to the guide for writing effective Gno code. This document is designed
+to help you understand the nuances of Gno and how to use it effectively.
+
+Before we dive in, it's important to note that Gno shares several similarities
+with Go. Therefore, if you haven't already, we highly recommend reading
+["Effective Go"](https://go.dev/doc/effective_go) as a primer.
+
+## Counter-Intuitive Good Practices
+
+This section highlights some Gno good practices that might seem
+counter-intuitive, especially if you're coming from a Go background.
+
+### Embrace Global Variables in Gno
+
+In Gno, using global variables isn't just acceptable - it's encouraged. This is
+because global variables in Gno provide a way to have persisted states
+automatically.
+
+In Go, you would typically write your logic and maintain some state in memory.
+However, to persist the state and ensure it survives a restart, you would need
+to use a store (like a plain file, custom file structure, a database, a
+key-value store, an API, etc.).
+
+In contrast, Gno simplifies this process. When you declare global variables in
+Gno, the GnoVM automatically persists and restores them as needed between each
+run.
+
+However, be mindful not to export your global variables. Doing so would make
```

Comment:
Is this still true? I think this code path it ends up following if something like that happens. https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/realm.go#L148

Reply:
You're likely correct.

I believe we should examine this more thoroughly, ensuring we assess not just the variable, but also aspects like whether it's a pointer.